### ISSUE TITLE
Optimize conversion of data.frame to matrix

### ISSUE THREAD

**Issue by ksens** [Paradigm4/revealgenomics#63]
@mmccorry Thanks for offering to look at optimizations in this API. 

This one could be a useful optimization to start with. The following can be repro-ed on the TCGA machine that we used for a recent demo. 

```R
rm(list=ls())
library(revealgenomics)
creds_file = '~/.scidb_secure_user_auth'
options(scidb.result_size_limit = 2048)
rg_connect(username = read_json(creds_file)$`user-name`, password = read_json(creds_file)$`user-password`)

id1 = find_dataset_id_by_grep("BRCA")
stopifnot(length(id1) == 1)

ms = search_measurementsets(dataset_id = id1)

biosample_ref = search_biosamples(dataset_id = id1)
system.time({
  res = search_expression(measurementset = ms[ms$name == 'HTSeq - FPKM-UQ', ], biosample_ref = biosample_ref)
})
# Downloading features to form ExpressionSet
# Forming ExpressionSet
# 76.750 seconds

format(object.size(exprs(res)), units = "MB")
# [1] "532.3 Mb"
```

As you can see, the download from scidb is quite sizable. However a lot of the time is spent in the "Forming ExpressionSet" part -- potentially at / around this line:
https://github.com/Paradigm4/revealgenomics/blob/8fa89bc80a7e8cbd5d991c339cd8d32245e615ef/R/functions.R#L1201

@apoliakov pointed out that he developed a fast data.frame to matrix conversion routine here:
https://github.com/Paradigm4/biobank/blob/e4a7782e4c55e2596335c936e3c339818c054b83/pkg/R/biobank.R#L2558

We might want to start using the same in this API. 

Also I had found the following in previous searches -- might be relevant: https://www.google.com/search?q=faster+reshape+r

**Comment by mmccorry** [comment]
In https://github.com/Paradigm4/revealgenomics/commit/8799f2c5052033f21bf19b0168e7856db5347f81, implemented a faster alternative to `acast()` using `data.table` and `Matrix`.

Had running successfully on a clone, failing in TravisCI.

**Comment by mmccorry** [comment]
Test script with output with updates from above commit:
```{r}
rm(list=ls())
gc()
library(revealgenomics)
library(tictoc)
creds_file = '~/.scidb_secure_user_auth'
options(scidb.result_size_limit = 2048)
rg_connect(username = read_json(creds_file)$`user-name`, password = read_json(creds_file)$`user-password`)

tic("finding dataset id...")
id1 = find_dataset_id_by_grep("BRCA")
stopifnot(length(id1) == 1)
toc()

tic("searching measurement sets...")
ms = search_measurementsets(dataset_id = id1)
toc()

tic("getting biosample ref by searching biosamples...")
biosample_ref = search_biosamples(dataset_id = id1)
toc()

tic("fetching the measurement set... ")
res <- search_expression(measurementset = ms[ms$name == 'HTSeq - FPKM-UQ', ], biosample_ref = biosample_ref)
toc()

print(format(object.size(exprs(res)), units = "MB"))
```

```
hostname was not provided. Connecting to localhost 
finding dataset id...: 0.101 sec elapsed
searching measurement sets...: 0.16 sec elapsed
getting biosample ref by searching biosamples...: 0.907 sec elapsed
Downloading full dataset from SciDB ...  done.
Downloading features to form ExpressionSet
Forming ExpressionSet
Reshaping expr to Matrix... done.
fetching the measurement set... : 44.378 sec elapsed
[1] "532.3 Mb"
````

(The part where it outputs "Downloading full dataset from SciDB..." takes roughly 30 seconds.)

**Comment by ksens** [comment]
@mmccorry 's optimizations are available in PR #100.
